### PR TITLE
Adsk Contrib - Update config default version and optimization enhancements

### PR DESCRIPTION
**Pull request by BernardLefebvre**
Default config version is now the latest version (today version 2).  Unit tests have also been adjusted to take this into account.  This implements issue #1087.

New optimization flags have been added:
- OPTIMIZATION_SIMPLIFY_OPS: break down certain ops into simpler ops when possible. Today, only CDL make use of this, and is replaced by a matrix when possible. The goal is to speed processing and allow more ops to be combined. The flag is enabled with the default optimization.

- OPTIMIZATION_FAST_LOG_EXP_POW: SSE implementation of log, power, and exponent is faster but less precise. When this new flag is set the fast implement is used, else the non-SSE implementation is used. The flag is enabled by default. This applies to several transforms including CDL, Exponent (v2), ExposureContrast, and Log.

Signed-off-by: Bernard Lefebvre <bernard.lefebvre@autodesk.com>

### CODE REVIEW

**Comment by zachlewis** [comment]
Does this mean that the only way to "create" a v1 config via the API is to load an existing default v1 config?

**Comment by BernardLefebvre** [comment]
> Does this mean that the only way to "create" a v1 config via the API is to load an existing default v1 config?

You can call Config::Create and then set the version. See tests/cpu/Config_tests.cpp line 5178

**Comment by doug-walker** [comment]
As usual this is a notice that we will merge this on Monday, if there are no more discussions on this PR.

File: include/OpenColorIO/OpenColorTypes.h [hunk 1]
```diff
@@ -537,11 +537,18 @@ enum OptimizationFlags : unsigned long
      */
     OPTIMIZATION_LUT_INV_FAST                    = 0x02000000,
 
+    // For CPU processor, in SSE mode, use a faster approximation for log, exp, and pow.
```
Comment (michdolan) [inline include/OpenColorIO/OpenColorTypes.h:540]: These new flags should also be added to the Python bindings:
https://github.com/AcademySoftwareFoundation/OpenColorIO/blob/master/src/bindings/python/PyTypes.cpp

**Review by michdolan — CHANGES_REQUESTED** [review]
One minor omissions. LGTM otherwise.

File: include/OpenColorIO/OpenColorTypes.h (hunk 1 above)
Comment (BernardLefebvre) [reply include/OpenColorIO/OpenColorTypes.h:540]: Added

Comment (michdolan) [reply include/OpenColorIO/OpenColorTypes.h:540]: Thanks!

**Review by michdolan — APPROVED** [review]

**Review by hodoulp — APPROVED** [review]

**Pull request merged**